usb_flash_drive_copy.py

```diff
@@ -0,0 +1,543 @@
+        if isinstance(subject, Button):
+            if not subject.is_active:
+                log.write("Shutdown OTCamera.")
+                self.shutdown()
+
+    def shutdown(self):
+        """Shutdown OTCamera."""
+
+        self._turn_off_all_leds()
+        self.rec_led.blink(n=4, background=False)
+
+        self._turn_off_all_leds()
+
+        if not config.DEBUG_MODE_ON:
+            log.closefile()
+            subprocess.call("sudo shutdown -h now", shell=True)
+
+    def _turn_off_all_leds(self):
+        """Turn off all LEDs."""
+        self.power_led.turn_off()
+        self.wifi_led.turn_off()
+        self.rec_led.turn_off()
+
+    def copy_to_usb(self, copy_info: CopyInformation) -> None:
+        """Copy over videos to USB flash drive.
+
+        The WiFi LED blinking indicates the videos being copied over.
+        The WiFi LED constantly being on indicates that the copy process is finished.
+
+        Args:
+            copy_info (CopyInformation): the copy information.
+        """
+        self.wifi_led.blink()
+        log.write("Start copying files")
+        for video in copy_info.videos:
+            if video.copied:
+                log.write(f"Video at: '{ video.path}' already copied. Skipping.")
+                continue
+
+            if not video.path.exists():
+                log.write(
+                    f"Video at: '{ video.path}' does not exists.", log.LogLevel.WARNING
+                )
+                continue
+
+            shutil.copy2(
+                src=video.path,
+                dst=copy_info.dest_dir / video.filename,
+            )
+            video.copied = True
+            log.write(f"Video: '{video.path}' copied.")
+            log.write(f"Video: '{video.path}' copied.")
+        log.write("Copying over videos to USB flash drive finished.")
+        self.wifi_led.turn_on()
+
+    def delete(self, copy_info: CopyInformation) -> None:
+        """Delete videos marked for deletion on OTCamera.
+
+        The recording LED blinking indicates the videos being deleted.
+        The recording LED constantly being on indicates that the delete process has
+        finished.
+
+        Args:
+            copy_info (CopyInformation): The copy information.
+        """
+        self.rec_led.blink()
+        for video in copy_info.videos.copy():
+            if not video.path.exists():
+                log.write(
+                    f"Video at: '{ video.path}' does not exist.", log.LogLevel.WARNING
+                )
+                copy_info.remove(video)
+                continue
+            if not video.delete:
+                log.write(
+                    f"Video at: '{ video.path}' not marked for deletion. Skipping.",
+                )
+                continue
+
+            if config.DEBUG_MODE_ON:
+                log.write("Debug mode on. Only mock deleting file.", log.LogLevel.DEBUG)
+            else:
+                video.path.unlink()
```

Comment:
Could there be an exception? See comment in line 400.

Reply:
Here a possible error could occur if another user logs into OTCamera and manually deleteing videos while the USB flash drive downloader is also trying to delete the same files. Will add a try-except clause when trying to delete the videos.